packages/@sanity/vision/src/components/QueryRecall.tsx

```diff
@@ -0,0 +1,430 @@
+
+          const isEdited = isSelected && !areQueriesEqual
+          return (
+            <Card
+              key={q._key}
+              width={'fill'}
+              padding={4}
+              border
+              tone={isSelected ? 'positive' : 'default'}
+              onClick={() => {
+                setSelectedUrl(q.url) // Update the selected query immediately
+                const parsedUrl = getStateFromUrl(q.url)
+                if (parsedUrl) {
+                  setStateFromParsedUrl(parsedUrl)
+                }
+              }}
+              style={{position: 'relative'}}
+            >
+              <Stack space={3}>
+                <Flex justify="space-between" align={'center'}>
+                  <Flex align="center" gap={2} paddingRight={1}>
+                    {editingKey === q._key ? (
+                      <TextInput
+                        value={editingTitle}
+                        onChange={(event) => setEditingTitle(event.currentTarget.value)}
+                        onKeyDown={(event) => {
+                          if (event.key === 'Enter') {
+                            handleTitleSave(q, editingTitle)
+                          } else if (event.key === 'Escape') {
+                            setEditingKey(null)
+                          }
+                        }}
+                        onBlur={() => handleTitleSave(q, editingTitle)}
+                        autoFocus
+                        style={{maxWidth: '170px', height: '24px'}}
+                      />
+                    ) : (
+                      <Text
+                        weight="bold"
+                        size={3}
+                        textOverflow="ellipsis"
+                        style={{maxWidth: '170px', cursor: 'pointer', padding: '4px 0'}}
+                        title={
+                          optimisticTitles[q._key] ||
+                          q.title ||
+                          q._key.slice(q._key.length - 5, q._key.length)
+                        }
+                        onClick={() => {
+                          setEditingKey(q._key)
+                          setEditingTitle(q.title || q._key.slice(0, 5))
+                        }}
+                      >
+                        {optimisticTitles[q._key] ||
+                          q.title ||
+                          q._key.slice(q._key.length - 5, q._key.length)}
+                      </Text>
+                    )}
+                    {isEdited && (
+                      <Box
+                        style={{
+                          width: '6px',
+                          height: '6px',
+                          borderRadius: '50%',
+                          backgroundColor: 'var(--card-focus-ring-color)',
+                        }}
+                      />
+                    )}
+                  </Flex>
+                  <MenuButton
+                    button={<EllipsisVerticalIcon />}
```

Comment:
This should preferably be a button, given users will need to click it and we want it to be focusable among having hover effect.
We are exporting a button from `sanity` that is useful for this. 

`import {ContextMenuButton} from "sanity"`
```suggestion
                    button={<ContextMenuButton />}
```